rpcs3/Crypto/unedat.cpp

```diff
@@ -715,26 +716,31 @@ bool extract_all_data(const fs::file* input, const fs::file* output, const char*
 
 		// Select EDAT key.
 		if ((NPD.license & 0x3) == 0x3)           // Type 3: Use supplied devklic.
-			memcpy(&key, devklic, 0x10);
-		else if ((NPD.license & 0x2) == 0x2)      // Type 2: Use key from RAP file (RIF key).
 		{
-			memcpy(&key, rifkey, 0x10);
+			std::memcpy(&key, devklic, 0x10);
+		}
+		else // Type 2: Use key from RAP file (RIF key). (also used for type 1 at the moment)
+		{
+			const std::string rap_path = rpcs3::utils::get_rap_file_path(NPD.content_id);
```

Comment:
I have concerns that this will make everything slower, since it will now look for a rap file everytime.

Reply:
DecryptEDAT has always read RAP file, and this function is only used by it.

Reply:
no. it was only reading a rap file if provided. now it always searches for the rap file, which it clearly wasn't doing before.

Reply:
for example package installation

Reply:
EDAT that needs RAP file will always need it, not providing RAP path in such case will just yield a decryption failure.
And looking for RAP file typically ends within a single iteration if RAP file exists, fs::is_file is not expensive.
If fs::is_file is really a concern get_rap_file_path can return fs::file to it as well as path.

Reply:
> https://github.com/RPCS3/rpcs3/pull/10934#discussion_r720637085
PKG installation currently only supports SDAT decrytion, which doesn't need RAP and won't reach this code.

Reply:
The way I see it is that it prevents decryption failure bugs, simplifies usage with very little performance panelty.

Reply:
Removed redundancy in verify_c00_unlock_edat because of it.

Reply:
We'll see if there are regressions ;)